### PR TITLE
[ADD] rpc: Go examples

### PR DESCRIPTION
**Pull request by mrhdias**
Examples rely on `kolo/xmlrpc` as well as on the standard library package `log` to panic on error, matching snippets in other languages (which tend to raise exceptions).

As with other languages, only the RPC interaction is spelled out.

### CODE REVIEW

File: content/developer/reference/external_api.rst [hunk 1]
```diff
@@ -303,6 +391,30 @@ Each call to ``execute_kw`` takes the following parameters:
...
+                 "raise_exception": false,
+             },
+         }, &result); err != nil {
+             log.Fatalln(err)
+         }
+         fmt.Println(result)
+
+         if err := models.Close(); err != nil {
+             log.Fatalln(err)
+         }
```
[the remark this answers is not in the document]
Comment (mrhdias) [reply content/developer/reference/external_api.rst]: Fixed

File: content/developer/reference/external_api.rst [hunk 2]
```diff
@@ -247,6 +319,22 @@ Result:
 
       int uid = (int)client.execute(common_config, "authenticate", asList(db, username, password, emptyMap()));
 
+   .. code-tab:: go
+
+      var uid int64
+
+      if err := client.Call("authenticate", []any{
```
[the remark this answers is not in the document]
Comment (mrhdias) [reply content/developer/reference/external_api.rst:306]: Fixed

File: content/developer/reference/external_api.rst [hunk 3]
```diff
@@ -792,6 +1055,40 @@ a record).
              asList(asList(id))
          )));
 
+      .. code-tab:: go
+
+         var result bool
```
[the remark this answers is not in the document]
Comment (mrhdias) [reply content/developer/reference/external_api.rst:994]: Fixed

**Comment by mrhdias** [comment]
> 
The treatment of return values is not inconsistent. If we know what is returned we use a structure, if we don't know we use an interface{} which is the same as using the "any" expression. But if you wish, I can pass everything to interface{}.

**Comment by mrhdias** [comment]
I think the examples are ok now. Check the latest revision. The Go examples closely follow the Java examples.

**Comment by AntoineVDV** [comment]
@robodoo delegate=xmo-odoo

**Comment by mrhdias** [comment]
> > I think the examples are ok now. Check the latest revision. The Go examples closely follow the Java examples.
> 
> The error handling was fine by me (most other languages would trigger exceptions on server-side error so the errors didn't need to he handled explicitely), I'm fine with approving as-is, or with adding the error checking stuff back in, as you prefer.

Thanks. I've added error checking again, so it's much clearer for developers to check for errors in each example. I followed this example from the go playground for errors.
https://go.dev/play/p/nPj4DTEuUnW

**Pull request closed without merging**